test/symmetric.jl

```diff
@@ -1035,6 +1035,16 @@ end
         @test Aherm isa Hermitian
         @test Aherm.uplo == LinearAlgebra.char_uplo(uplo)
     end
+    @testset "hermitianpart for numbers" begin
+    @test hermitianpart(3 + 4im) == 3
+    @test hermitianpart(5) == 5.0
+    @test typeof(hermitianpart(5)) == Float64
+    @test hermitianpart(2.5) == 2.5
+    @test hermitianpart(-1 + 0im) == -1
+    @test typeof(hermitianpart(3 + 4im)) == 3.0
+    @test typeof(hermitianpart(3 + 4im)) == Float64
+    @test typeof(hermitianpart(2.5 + 3im)) == Float64
```

Comment:
```suggestion
    @test hermitianpart(2.5) == 2.5
    @test hermitianpart(-1 + 0im) == -1
    @test typeof(hermitianpart(3 + 4im)) == 3.0
```

These type tests wouldn't be needed anymore.